### ISSUE TITLE
Clear the datasource. API method to reload the scroller

### ISSUE THREAD

**Issue by cristigav** [dhilt/ngx-ui-scroll#34]
I tried to use the scroll with a datasource that is the result of a search, but I can not find a way to tell the scroll component that the datasource has changed in content, so it only triggers the get method first time.

I have tried setting my datasource to null and then create a new one after a new search, and also tried to put an *ngIf on the parent div to disable and re-enable it again but still the scroll does not care if my datasource has changed.

Is there a way I can clear the datasource?

**Comment by dhilt** [comment]
@cristigav Hey! I think you need a reload method... Currently there is no external API that can be used to manipulate and assess the scroller. But there is time to change it, and I will implement `reload()` method somewhere on Datasource for next (v0.0.3) release.

It will take some time. As a workaround I can suggest following approach (by your idea based on complete rebuild of the scroller via ngIf):

```
export class MyComponent  {
  public reload = true;
  doReload() {
    this.reload = !this.reload;
    setTimeout(() => this.reload = !this.reload);
  }
  // datasource etc
}
```

And the template

```
<div class="viewport" *ngIf="reload">
  <div *uiScroll="let item of datasource"><!-- ... --></div>
</div>
```

Just trigger `MyComponent.reload()` when you want the scroller to be re-built after the data is changed. I created a complex demo on stackblitz for you can try this approach in action: https://stackblitz.com/edit/angular-gfqsft

**Comment by cristigav** [comment]
Thanks. The workaround works. The problem with the workaround is that when I am loading the next page and do a reload, the scroll will start from the beginning. So I will wait for a reload method :)

**Comment by dhilt** [comment]
@cristigav v0.0.4-rc has been just released and available via npm with "next" tag:

```
npm install ngx-ui-scroll@next
```

In this version the Datasource is endowed with a special Adapter property which has the reload method:

```
this.datasource.adapter.reload(index);
```

It allows you to reload the scroller to a specific position without destroying the viewport. I also added a  reload sample to the demo App: https://dhilt.github.io/ngx-ui-scroll/#/adapter#reload.

The work is not finished, some tests need to be added to the project, and while I'm implementing them, you may try `ngx-ui-scroll@next` in your App. Would be great to get feedback!

**Comment by dhilt** [comment]
Closed per v0.0.4 is released.
